test_common.py

```diff
@@ -0,0 +1,219 @@
+# functionalities and invariance, e.g. SYMMETRIC_METRICS lists all metrics that
+# are symmetric with respect to their input argument y_true and y_pred.
+#
+
+# Symmetric with respect to their input arguments y_true and y_pred
+# metric(y_true, y_pred) == metric(y_pred, y_true).
+SYMMETRIC_METRICS = [ "adjusted_rand_score" , "v_measure_score" ,
+    "mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score",
+    ]
+                    
+NON_SYMMETRIC_METRICS = ["homogeneity_score" , "completeness_score"]
+                    
+#METIRICS with output between 0 and 1 and belong to BICLUSTER METRICS
+METRICS_NORMALIZED_OUTPUT_BICLUSTER = ["consensus_score","jaccard"]
+ 
+#metrics with output between 0 and 1                                      
+METRICS_NORMALIZED_OUTPUT = ["adjusted_rand_score","homogeneity_score",
+    "completeness_score","v_measure_score",
+    "adjusted_mutual_info_score","normalized_mutual_info_score"
+    ]                  
+                                
+#when information is zero these metrics output zero
+METRICS_ZERO_INFO = ["normalized_mutual_info_score","v_measure_score",
+    "adjusted_mutual_info_score"
+    ]
+                      
+#METRICS where permutations oflabels dont change score
+METRICS_PERMUTE_LABELS = ["homogeneity_score","v_measure_score",
+    "completeness_score","mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score"
+    ]
+                        
+#metrics which result in 0 when a class is split across different clusters
+ClASS_BASED_METRICS = ["adjusted_mutual_info","normalized_mutual_info",
+    "v_measure_score"
+    ]
+                        
+
+def test_symmetry():
+    #symmetric_metrics
+    for name in SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,0],[42,7,42]),
+                            metric([42,7,42], [0,1,0]),
+                            err_msg="%s is not symmetric" % name)
+            
+            
+        assert_almost_equal(metric([0,0,1,1,2,2],[1,1,3,4,2,1]),
+                            metric([1,1,3,4,2,1],[0,0,1,1,2,2]),
+                            err_msg="%s is not symmetric" % name)
+        assert_almost_equal(metric([0,1,2,5,4,9],[0,1,9,4,3,5]),
+                            metric([0,1,9,4,3,5],[0,1,2,5,4,9]),
+                            err_msg="%s is not symmetric" % name)
+                             
+    for name in NON_SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,2,5,4,9],[0,1,9,4,3,5]),
+                            metric([0,1,9,4,3,5],[0,1,2,5,4,9]),
+                            err_msg="%s is not symmetric" % name)
+                            
+      
+                    
+#test function for metrics whose output in range 0 to 1
+#bicluster metrics have different input format so they are handled differently                  
+def test_normalized_output_bicluster():

```

Comment:
This does not belong here. it does not test a common invariance across multiple metrics. Jaccard is not a biclustering metric. It does not evaluate a pair of clusterings but a pair of clusters. It should not be present in `test_common.py` at all.

It's fine if test_common.py does not handle any biclustering.